### DISCUSSION TITLE
How to manually focus input inside a predefined useFieldArray

### DISCUSSION THREAD

**Discussion by foubei** [react-hook-form/react-hook-form#1775] [General] [answered]
I have a verification code input which is separated on 6 inputs. I am using useFieldArray to create my 6 inputs and every input is declared as <Controller>. But how can I focus an input referenced in its controller?

My scenario is that I have to focus the next input everytime the current focused input its value is entered by the user.

PS: I am using functional components using useFormContext and useFieldArray hooks

![Screenshot_1591266497](https://user-images.githubusercontent.com/12202571/83746127-36517c00-a691-11ea-81b8-ff6b033e0552.png)

**Comment by bluebill1049** [comment, marked as answer]
i would just use `useForm` and store all of their `ref`s,  https://react-hook-form.com/faqs#Howtosharerefusage

**Reply by foubei** [reply]
Thanks for your reply. Does Controller has a ref property? and how can be used

**Reply by bluebill1049** [reply]
Controller doesn't have ref, you will have to get the ref from the component that you wrapped with.

**Reply by foubei** [reply]
So I get every input ref from input itself and push to refs array. I was trying the controller ref property and wasn't able to resolve it.

Thanks for your help. Now I can make it working

**Reply by foubei** [reply]
Why you suggested to use useForm instead of useFieldArray? what's the big difference between both?

PS: I was using redux-form and just switched to RHF few days ago

**Reply by bluebill1049** [reply]
because useFieldArray is for dynamic inputs, by the look your form is static.

**Reply by foubei** [reply]
Got it. appreciate tour help
